### PR TITLE
scmpuff: init

### PR DESCRIPTION
**Pull request by cpcloud**
### Description

This PR adds the [`scmpuff`](https://github.com/mroth/scmpuff) tool to home-manager.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [x] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: .github/CODEOWNERS
Comment (sumnerevans) [inline .github/CODEOWNERS:139]: Go ahead and add yourself as the codeowner for the tests as well:

```suggestion
/modules/programs/scmpuff.nix                         @cpcloud
/tests/modules/programs/scmpuff                       @cpcloud
```

File: tests/modules/programs/scmpuff/bash.nix
Comment (sumnerevans) [inline tests/modules/programs/scmpuff/bash.nix:4]: I think it may be worth adding tests for when `enableBashIntegration` and `enableZshIntegration` are `false`, but I will defer to a maintainer.

**Review by sumnerevans — CHANGES_REQUESTED** [review]

File: modules/programs/scmpuff.nix
Comment (berbiche) [inline modules/programs/scmpuff.nix:9]: `mkEnableOption` expands the description to `Whether to enable ${x}.`, where `x` is the string you passed as argument.

Also note that the final period is already present.

```suggestion
    enable = mkEnableOption
      "scmpuff, a command line tool that allows you to work quicker with Git by substituting numeric shortcuts for files";
```

**Review by berbiche — CHANGES_REQUESTED** [review]
Thanks for the contribution!

One minor detail to change and it's good to merge :)

File: tests/modules/programs/scmpuff/bash.nix
Comment (cpcloud) [reply tests/modules/programs/scmpuff/bash.nix:4]: Done.

**Review by sumnerevans — APPROVED** [review]

**Comment by cpcloud** [comment]
@berbiche @sumnerevans Can one of y'all merge this PR?

**Comment by sumnerevans** [comment]
@cpcloud I don't have commit access :)

**Review by berbiche — APPROVED** [review]

**Pull request merged**
